### DISCUSSION TITLE
Performance issue while dragging component to canvas when you have 2k component on canvas

### DISCUSSION THREAD

**Discussion by mohdSuhailCoditation** [GrapesJS/grapesjs#4321] [Q&A]
If there are 2K components on canvas then dragging a block on to the canvas lags the green line.
**What browser are you using?**
Google Chrome Version 101.0.4951.41 (Official Build) (64-bit)
**Demo**
https://user-images.githubusercontent.com/100282281/167993605-a31e7ffb-f594-4795-a8d5-fe8dac11c154.mp4

https://user-images.githubusercontent.com/100282281/167994336-856c08ba-2f84-4019-be8e-cfd6d9003b42.mp4

**Explanation**
I have added 2k components on the canvas. Then I am trying to drag a block on the canvas & it is showing a huge lag.
Script which I am using to populate 2k components: 
`const populate = () => {
    for(let i = 0;i<2000;i++)
    editor.addComponents(
      '<div>ABC</div>'
    );
  }`

how can i solve this issue? there is a way to improve the performance having a lot of blocks on canvas?

**Comment by artf** [comment]
I'll close this one in favor of the issue https://github.com/artf/grapesjs/issues/4323
